src/main/java/com/google/api/codegen/transformer/java/JavaModelTypeNameConverter.java

```diff
@@ -146,6 +146,9 @@ private TypeName getTypeNameForElementType(TypeRef type, boolean shouldBoxPrimit
     }
     switch (type.getKind()) {
       case TYPE_MESSAGE:
+        if (type.isMap() && type.isMessage() && type.isRepeated()) {
```

Comment:
Probably, but I was not sure, so made it as rigorous as possible.

Reply:
I confirmed by checking the implementation of the `isMessage()` and `isRepeated()` and `isMap()` method that if isMap() is true than the other two are guaranteed to be true as well. Removed the other method calls since they are redundant.